circuits/src/zk_circuits/valid_commitments.rs

```diff
@@ -0,0 +1,950 @@
+        let witness_var = witness_commitment.commit_verifier(&mut verifier).unwrap();
+        let statement_var = statement.commit_public(&mut verifier).unwrap();
+
+        // Apply the constrains
+        Self::circuit(statement_var, witness_var, &mut verifier);
+
+        // Verify the proof
+        let bp_gens = BulletproofGens::new(Self::BP_GENS_CAPACITY, 1 /* party_capacity */);
+        verifier
+            .verify(&proof, &bp_gens)
+            .map_err(VerifierError::R1CS)
+    }
+}
+
+// ---------
+// | Tests |
+// ---------
+
+#[cfg(test)]
+mod test {
+    #![allow(non_snake_case)]
+
+    use curve25519_dalek::scalar::Scalar;
+    use lazy_static::lazy_static;
+    use merlin::Transcript;
+    use mpc_bulletproof::{r1cs::Prover, PedersenGens};
+    use num_bigint::BigUint;
+    use rand::{thread_rng, Rng};
+    use rand_core::OsRng;
+
+    use crate::{
+        types::{
+            balance::{Balance, BalanceSecretShare},
+            fee::FeeSecretShare,
+            order::{OrderSecretShare, OrderSide},
+        },
+        zk_circuits::test_helpers::{
+            create_wallet_shares, create_wallet_shares_from_private, SizedWallet, INITIAL_WALLET,
+            MAX_BALANCES, MAX_FEES, MAX_ORDERS,
+        },
+        CommitPublic, CommitWitness,
+    };
+
+    use super::{ValidCommitments, ValidCommitmentsStatement, ValidCommitmentsWitness};
+
+    // --------------
+    // | Dummy Data |
+    // --------------
+
+    lazy_static! {
+        /// A wallet needing augmentation to receive its side of an order
+        static ref UNAUGMENTED_WALLET: SizedWallet = {
+            // Zero the wallet balance corresponding to the received mint
+            let mut wallet = INITIAL_WALLET.clone();
+            let order_receive_mint = match wallet.orders[0].side {
+                OrderSide::Buy => wallet.orders[0].base_mint.clone(),
+                OrderSide::Sell => wallet.orders[0].quote_mint.clone(),
+            };
+
+            // Find the received mint balance
+            let (balance_ind, _) = find_balance(order_receive_mint, &mut wallet, false /* augment */);
+            wallet.balances[balance_ind] = Balance::default();
+            wallet
+        };
+    }
+
+    /// A type alias for the VALID COMMITMENTS witness with size parameters attached
+    type SizedWitness = ValidCommitmentsWitness<MAX_BALANCES, MAX_ORDERS, MAX_FEES>;
+
+    // -----------
+    // | Helpers |
+    // -----------
+
+    /// Construct a valid witness and statement from the given wallet
+    ///
+    /// Simply chooses a random order to match against from the wallet
+    fn create_witness_and_statement(
+        wallet: &SizedWallet,
+    ) -> (SizedWitness, ValidCommitmentsStatement) {
+        let mut rng = thread_rng();
+
+        // Split the wallet into secret shares
+        let (private_shares, public_shares) = create_wallet_shares(wallet);
+
+        // Choose an order and fee to match on
+        let ind_order = 0;
```

Comment:
Makes sense, I originally had it as random, but it makes it harder to predictably clobber invalid values in the test cases below. I'll leave it as static for now just to avoid complicating all the tests